src/test/mir-opt/const-promotion-extern-static/rustc.BAR.PromoteTemps.diff

```diff
@@ -0,0 +1,64 @@
+-                                          // + span: $DIR/const-promotion-extern-static.rs:8:41: 8:42
+-                                          // + literal: Const { ty: &i32, val: Value(Scalar(alloc0+0)) }
+-         _4 = &(*_5);                     // bb0[6]: scope 0 at $DIR/const-promotion-extern-static.rs:8:40: 8:42
+-         _3 = [move _4];                  // bb0[7]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+-         _2 = &_3;                        // bb0[8]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+-         _1 = move _2 as &[&'static i32] (Pointer(Unsize)); // bb0[9]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
+-         _0 = const core::slice::<impl [&'static i32]>::as_ptr(move _1) -> [return: bb2, unwind: bb1]; // bb0[10]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:52
++                                          // + span: $DIR/const-promotion-extern-static.rs:8:39: 8:43
++                                          // + literal: Const { ty: &[&'static i32; 1], val: Unevaluated(DefId(0:6 ~ const_promotion_extern_static[317d]::BAR[0]), [], Some(promoted[0])) }
++         _2 = &(*_6);                     // bb0[3]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
++         _1 = move _2 as &[&'static i32] (Pointer(Unsize)); // bb0[4]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:43
++         _0 = const core::slice::<impl [&'static i32]>::as_ptr(move _1) -> [return: bb2, unwind: bb1]; // bb0[5]: scope 0 at $DIR/const-promotion-extern-static.rs:8:39: 8:52
+                                           // ty::Const
+                                           // + ty: for<'r> fn(&'r [&'static i32]) -> *const &'static i32 {core::slice::<impl [&'static i32]>::as_ptr}
+                                           // + val: Value(Scalar(<ZST>))
+                                           // mir::Constant
+                                           // + span: $DIR/const-promotion-extern-static.rs:8:44: 8:50
+                                           // + literal: Const { ty: for<'r> fn(&'r [&'static i32]) -> *const &'static i32 {core::slice::<impl [&'static i32]>::as_ptr}, val: Value(Scalar(<ZST>)) }
+      }
+  
+      bb1 (cleanup): {
+          resume;                          // bb1[0]: scope 0 at $DIR/const-promotion-extern-static.rs:8:1: 8:53
+      }
+  
+      bb2: {
+-         StorageDead(_5);                 // bb2[0]: scope 0 at $DIR/const-promotion-extern-static.rs:8:51: 8:52
+-         StorageDead(_3);                 // bb2[1]: scope 0 at $DIR/const-promotion-extern-static.rs:8:51: 8:52
+-         return;                          // bb2[2]: scope 0 at $DIR/const-promotion-extern-static.rs:8:1: 8:53
++         return;                          // bb2[0]: scope 0 at $DIR/const-promotion-extern-static.rs:8:1: 8:53
+      }
+  }
+- 
+- alloc0: DefId(0:5 ~ const_promotion_extern_static[317d]::Y[0]): 4 bytes, alignment 4
+- 2a 00 00 00 
```

Comment:
Is there some way to make it more obvious that the random sequence of characters here is the allocation content? That is not fully clear right now, as just
```
alloc0: DefId(0:5 ~ const_promotion_extern_static[317d]::Y[0]): 4 bytes, alignment 4
```
also seems like a self-contained useful chunk of information.

Reply:
IMO:
1. avoid using `{:?}` + `DefId`, use `def_path_str`(`_with_substs`) instead
2. change `N bytes, alignment M` to be less asymmetric
3. indent the byte contents and wrap them in `{...}`
4. make sure the bytes split onto multiple lines (like a hex editor) when there's a lot of them

Reply:
done